### PR TITLE
feat: SEARCH-1541 - Change FreeDiskSpace to use native binding

### PR DESCRIPTION
**Pull request by keerthi16**
## Description
Remove FreeDiskSpace which was dependant on .Net
And pass minimum disk space from SFE
[SEARCH-1541](https://perzoinc.atlassian.net/browse/SEARCH-1541)

## Approach
[comment]: # (How does this change address the problem?)
#### Problem with the code:
- N/A

#### Fix:
- Passing the minimum disk space required for swift-search from SFE as arg.

## Learning
N/A

#### Blog Posts
N/A

## Related PRs
[comment]: # (List the related PRs against other branches.)

Repo | Branch | PR #
------ | ------ | ------
SFE | branch-name | [#15257](https://github.com/SymphonyOSF/SFE-Client-App/pull/15257)
SFE-Context-Isolation | branch-name | [#8](https://github.com/KiranNiranjan/SFE-Client-App/pull/8)
SymphonyElectron | branch-name | [#692](https://github.com/symphonyoss/SymphonyElectron/pull/692)

## Open Questions if any and Todos
[comment]: # (When resolved, check the box and explain the answer.)
- [ ] Automation-Tests
- [x] Documentation
- [x] Language Translations
- [x] Unit-Tests

## Custom branch on PR builder
N/A

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -38,12 +38,10 @@
     "typescript": "2.9.2"
   },
   "dependencies": {
+    "diskusage": "1.1.2",
```
Comment (VishwasShashidhar) [inline package.json]: This library is more like a native addon isn't it? Does this work against Node 12?

File: src/search.ts [hunk 2]
```diff
@@ -25,15 +32,18 @@ export default class Search extends SearchUtils implements SearchInterface {
     private isInitialized: boolean;
     private isRealTimeIndexing: boolean;
     private readonly collector: any;
+    private readonly searchPeriodSubtract: number;
 
     /**
      * Constructor for the SymphonySearchEngine library
      * @param userId (for the index folder name)
      * @param key
```
Comment (VishwasShashidhar) [inline src/search.ts]: Please add description of the argument and what it does.

File: src/search.ts [hunk 3]
```diff
@@ -25,15 +32,18 @@ export default class Search extends SearchUtils implements SearchInterface {
...
+     * @param payload
```
Comment (VishwasShashidhar) [inline src/search.ts]: Please add description of the argument and what it does.

File: package.json (hunk 1 above)
Comment (keerthi16) [reply package.json]: @VishwasShashidhar Yes, it supports Node 12.

**Comment by keerthi16** [comment]
@VishwasShashidhar Update the PR please review. Thanks

**Comment by keerthi16** [comment]
Published `swift-search@1.55.2-beta.7`

**Pull request merged**